internal/backends/python/python.go

```diff
@@ -769,20 +762,84 @@ func makePythonUvBackend() api.LanguageBackend {
 
 		return pkgs
 	}
-	addExtraIndexes := func(pkgName string) {
-		extraIndexes, ok := extraIndexMap[pkgName]
-		if ok {
-			uvIndex := os.Getenv("UV_INDEX")
-
-			for _, index := range extraIndexes {
-				if strings.HasPrefix(runtime.GOOS, index.os) {
-					uvIndex = index.url + " " + uvIndex
+	addTorchOverride := func() error {
+		if !util.TomlEditorIsAvailable() {
+			return errors.New(
+				"toml-editor is not on the PATH, please install it with " +
+					"`nix profile install github:replit/toml-editor` or " +
+					"`cargo install --git https://github.com/replit/toml-editor` and ensure it's on the PATH.",
+			)
+		}
+
+		// check if the torch cpu index is already present
+		getOp := util.TomlEditorOp{
+			Op:   "get",
+			Path: "tool/uv/index",
+		}
+		response, err := util.ExecTomlEditor("pyproject.toml", []util.TomlEditorOp{getOp})
+		if err != nil {
+			return err
+		}
+		if len(response.Results) != 1 {
+			return fmt.Errorf("expected one result")
+		}
+		result := response.Results[0]
+		if result != nil {
+			if arr, ok := result.([]interface{}); ok {
+				if slices.ContainsFunc(arr, func(val interface{}) bool {
+					gotValue, ok := val.(map[string]interface{})
+					if !ok {
+						return false
+					}
+
+					url, ok := gotValue["url"]
+					if !ok {
+						return false
+					}
+
+					return url == "https://download.pytorch.org/whl/cpu"
+				}) {
+					return nil
 				}
 			}
+		}
+
+		value := map[string]interface{}{"name": "pytorch-cpu", "url": "https://download.pytorch.org/whl/cpu", "explicit": true}
+		valueBytes, err := json.Marshal(value)
+		if err != nil {
+			return err
+		}
+		addOp := util.TomlEditorOp{
+			Op:              "add",
+			TableHeaderPath: "tool/uv/index/[[]]",
+			Value:           string(valueBytes),
+		}
+		_, err = util.ExecTomlEditor("pyproject.toml", []util.TomlEditorOp{addOp})
+		if err != nil {
+			return err
+		}
 
-			os.Setenv("UV_INDEX", uvIndex)
+		for _, name := range torchOverrides {
+			value := []map[string]interface{}{
+				{"index": "pytorch-cpu", "marker": "platform_system == 'Linux'"},
+			}
+			valueBytes, err := json.Marshal(value)
```

Comment:
We need to query the `pyproject.toml` `[tool.uv.sources]` table to see if each package in `torchOverrides` actually has a pinned value, or if we need to add it.

The reason being, if we need to add more pins in the future we should be able to operate with files that have never been modified, or ones that we've touched with a previous version of this approach.

Reply:
Got it, did something similar to what you suggested above and tested adding a new torch override which seems to work!